### PR TITLE
Set up Heroku hosting, automatic deploys, and CI testing pipelines

### LINKED ISSUE #123: CORE: Setup web hosting and app deployment
Figure out which hosting service to use and deploy the app using the hosting service. 

This issue will be useful for testing user location tracking.

### PR DESCRIPTION
**Pull request by gaskij**
## Description
This PR includes the code to set up GitHub Actions to provide automatic testing, linting, and build checks when pushing code to any branch or creating a PR targeting master. This also sets up the basis for having automatic deployment to Heroku.

Fixes #116, fixes #123

## Solution
I created the `.github/workflows/main.yaml` file to set up GitHub Actions, and created the app and pipeline on Heroku to go along with it. The app can be found here: https://rpicampusmap.herokuapp.com/

## Known Issues
There are no issues caused by this PR, but there is a change to the linter. It now expects LF (Unix) linebreaks instead of CRLF (Windows).

## Testing Procedures
* Visit the deployed website and ensure it is running
* Become members of the team on Heroku and become familiar with the pipeline

## Checklist for author
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if any)
- [x] My changes generate no new warnings

### CODE REVIEW

File: Procfile [hunk 1]
```diff
@@ -0,0 +1 @@
+web: node api/dist/server.js
```
Comment (saprap1) [inline Procfile:1]: missing newline

File: campusmap/.eslintrc.json [hunk 2]
```diff
@@ -1,68 +1,77 @@
...
+      "always"
+    ],
+    "file-extension": "off",
+    "import/extensions": "off",
+    "import/no-extra-dependencies": "off",
+    "import/no-unresolved": "off",
+    "no-unused-vars": "off",
+    "prefer-destructuring": "off"
+  }
 }
```
Comment (saprap1) [inline campusmap/.eslintrc.json:77]: missing newline

File: campusmap/jest.config.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  preset: 'ts-jest',
+  testEnvironment: 'node',
+};
```
Comment (saprap1) [inline campusmap/jest.config.js:4]: missing newline

File: campusmap/src/search-results/SearchResultsPage.test.tsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+const sum = (a: number, b: number) => a + b;
+
+describe('<SearchResultsPage />', () => {
```
Comment (saprap1) [inline campusmap/src/search-results/SearchResultsPage.test.tsx:3]: is this just meant to be an example test case? I don't really see how this tests/relates to the Search Results Page directly.

Comment (gaskij) [reply campusmap/src/search-results/SearchResultsPage.test.tsx:3]: Yes, this is just an example to test that the test runner on GitHub actions works. The actual tests for the pages will be written soon.

**Review by saprap1 — APPROVED** [review]

**Pull request merged**